docs/book/src/getting-started/testdata/project/internal/controller/memcached_controller_test.go

```diff
@@ -83,30 +82,21 @@ var _ = Describe("Memcached Controller", func() {
 			})
 			Expect(err).NotTo(HaveOccurred())
 			By("Checking if Deployment was successfully created in the reconciliation")
-			Eventually(func() error {
+			Eventually(func(g Gomega) {
 				found := &appsv1.Deployment{}
-				return k8sClient.Get(ctx, typeNamespacedName, found)
-			}, time.Minute, time.Second).Should(Succeed())
+				g.Expect(k8sClient.Get(ctx, typeNamespacedName, found)).To(Succeed())
+			}).Should(Succeed())
 
 			By("Checking the latest Status Condition added to the Memcached instance")
-			Eventually(func() error {
-				if memcached.Status.Conditions != nil &&
```

Comment:
I just checked that this test is broken by design; the test verifies that the status of the *created object*  in line 61.  And it does not return an error if the conditions are missing.  I'm hoping the test passes now that I have rewritten it to verify that the _server_ copy has correct status.

Reply:
I have tried to get this test to work, but to me it seems that the reconciler is not actively running, it only runs reconcile once, and it probably needs a few more reconciles for it to become "healthy".

I think I would need to run through the reconcile function again:

```
_, err := busyboxReconciler.Reconcile(ctx, reconcile.Request{
	NamespacedName: typeNamespacedName,
})
Expect(err).NotTo(HaveOccurred())
```

It's clear that the Conditions were never actually checked.  What should I do here?  Run the reconcile manually a few more times?